src/Simulation/CSharpGeneration/RewriteStep.fs

```diff
@@ -61,9 +48,31 @@ type Emitter() =
                 if content <> null then CompilationLoader.GeneratedFile(source, dir, ".dll.g.cs", content) |> ignore
 
             if not compilation.EntryPoints.IsEmpty then
-                let callable = context.allCallables.[Seq.exactlyOne compilation.EntryPoints]
-                let content = EntryPoint.generate context callable
-                CompilationLoader.GeneratedFile(callable.SourceFile, dir, ".EntryPoint.g.cs", content) |> ignore
+
+                let entryPointCallables =
+                    compilation.EntryPoints
+                    |> Seq.map (fun ep -> context.allCallables.[ep])
+
+                let entryPointSources =
+                    entryPointCallables
+                    |> Seq.groupBy (fun ep -> ep.Source.CodeFile)
+
+                // ToDo: remove commented out code
```

Comment:
Yes, I am aware of this, and will remove the commented out code at the next push.

Reply:
removed